_build_backend/vllm.py

```diff
@@ -0,0 +1,195 @@
+# SPDX-License-Identifier: Apache-2.0
+
+import importlib
+import os
+import sys
+from shutil import which
+from textwrap import dedent
+
+from setuptools.build_meta import *
+
+
+def load_module_from_path(module_name, path):
+    spec = importlib.util.spec_from_file_location(module_name, path)
+    module = importlib.util.module_from_spec(spec)
+    sys.modules[module_name] = module
+    spec.loader.exec_module(module)
+    return module
+
+
+ROOT_DIR = os.path.dirname(os.path.dirname(__file__))
+
+# cannot import envs directly because it depends on vllm, which is not installed yet
+envs = load_module_from_path('envs', os.path.join(ROOT_DIR, 'vllm', 'envs.py'))
+VLLM_TARGET_DEVICE = envs.VLLM_TARGET_DEVICE
+VLLM_USE_PRECOMPILED = envs.VLLM_USE_PRECOMPILED
+
+
+def _get_requires_for_build_extensions() -> list[str]:
```

Comment:
are the functions starting with "_" going to change soon? if not, i'd drop them, since this area already seems only for internal use.

Reply:
They might, but the main reason I added the leading `_` is to avoid confusion with other functions which are actual setuptools overrides such as `get_requires_for_build_wheel`